client/blocks/inline-help/inline-help-search-results.jsx

```diff
@@ -121,38 +172,47 @@ class InlineHelpSearchResults extends Component {
 		return selectedLink.link;
 	}
 
-	componentDidMount() {
-		this.props.setSearchResults( '', this.getContextResults() );
-	}
-
-	componentWillUpdate( nextProps ) {
-		if ( nextProps.shouldOpenSelectedResult ) {
-			const url = this.getSelectedUrl();
-			this.followHelpLink( url )();
-			this.props.didOpenResult();
-			window.location = url;
-			return;
-		}
-	}
-
-	followHelpLink = ( url ) => {
-		const payload = {
-			searchQuery: this.props.searchQuery,
-			currentUrl: window.location.href,
-			resultUrl: url,
-		};
+	followHelpLink = ( url, index ) => {
+		const railcar = this.getRailcarForIndex( index );
 		return () => {
-			this.props.recordTracksEvent( 'calypso_inlinehelp_link_open', payload );
+			// TrainTracks interact event
+			this.props.recordTracksEvent( 'calypso_traintracks_interact', {
+				action: 'calypso_inlinehelp_link_open',
+				railcar: railcar,
+			} );
+			// regular Tracks event
+			this.props.recordTracksEvent( 'calypso_inlinehelp_link_open', {
+				search_query: this.props.searchQuery,
+				current_url: window.location.href,
+				result_url: url,
+			} );
 		};
 	}
 
 	renderHelpLink = ( link, index ) => {
+		// TrainTracks render event
```

Comment:
Wouldn't an array of [`<TrackComponentView />`s](https://github.com/Automattic/wp-calypso/blob/50c55bf7285455b9595ebb2f5746d5ad2064e332/client/lib/analytics/track-component-view/README.md) simplify this and avoid the need for the custom event queue?

Reply:
Thanks @jblz, I hadn't seen that component — great suggestion! Solved in 1a45a4c.

Reply:
Are we not sending them as the render takes place because it slows down the rendering? If that is the case seems like there should be a generic Tracks queue that gets used.

Reply:
@jblz @gibrown thanks for your comments! 

We're not sending the events during `render` because sending the Tracks events apparently changes component state in some component somewhere. That makes React generate an error / warning because state isn't supposed to be changed during render. The `TrackComponentView` component solves this by shifting the event sending to its own `componentWillMount`.

Maybe there's a need for a generic Tracks queue anyhow? Even if that is the case I don't think that should be done in this PR.

Reply:
Ya a separate PR for that makes sense.